vm/src/obj/objdict.rs

```diff
@@ -295,29 +215,43 @@ impl PyDictRef {
             }
         }
     }
+
+    /// Take a python dictionary and convert it to attributes.
+    pub fn to_attributes(self) -> PyAttributes {
+        let mut attrs = PyAttributes::new();
+        for (key, value) in self.get_key_value_pairs() {
+            let key = objstr::get_value(&key);
+            attrs.insert(key, value);
+        }
+        attrs
+    }
 }
 
 impl DictProtocol for PyDictRef {
     fn contains_key(&self, k: &str) -> bool {
-        content_contains_key_str(&self.entries.borrow(), k)
+        self.entries.borrow().get(k).is_some()
     }
 
     fn get_item(&self, k: &str) -> Option<PyObjectRef> {
-        content_get_key_str(&self.entries.borrow(), k)
+        match self.entries.borrow().get(k) {
```

Comment:
For this situation it is ideal to use the map function of the Option type like this:
```rust
self.entries.borrow().get(k).map(|v| v.1.clone())
```

Reply:
Will do.